server/auth/serviceaccount/claims.go

```diff
@@ -60,3 +67,55 @@ func ClaimSetFor(restConfig *rest.Config) (*types.Claims, error) {
 
 	return claims, nil
 }
+
+func ClaimSetWithX509(restConfig *rest.Config) (*types.Claims, error) {
+	var cert *x509.Certificate
+
+	if len(restConfig.CertData) > 0 {
```

Comment:
reading on [TLSClientConfig](https://github.com/kubernetes/client-go/blob/v0.33.2/rest/config.go#L234), wouldn't CertData be nested under Config.TLSClientConfig